src/shared/Core/Interop/Windows/WindowsCredentialManager.cs

```diff
@@ -339,6 +345,11 @@ private string CreateTargetName(string service, string account)
                 sb.Append(serviceUri.Host);
             }
 
+            if (!serviceUri.IsDefaultPort)
+            {
+                sb.AppendFormat(":{0}", serviceUri.Port);
+            }
+
             if (!string.IsNullOrWhiteSpace(serviceUri.AbsolutePath.TrimEnd('/')))
             {
                 sb.Append(serviceUri.AbsolutePath);
```

Comment:
@ldennington the failing unit test case fails because of this (arguable) error in the code. Today we only effectively trim the trailing '/' if there is no path in the URI.

I.e,

```text
https://example.com/ ==> https://github.com
https://example.com/path/ ==> https://github.com/path/
```

I think we should probably be doing as the test is looking for (to always trim), but for people who have stored creds against the full path already (using `credential.useHttpPath=true`, we will be storing/updating a second credential (one without a slash) and then depending on the _order_ we look up existing credentials, match against _both_.

This may leave users with expired creds that we are not deleting but keep recalling.

What do you think we should do here?

Reply:
Spoke offline.. updated as per new revision and commit message:

```text
We also now always trim any trailing '/' from the end of the path of a
service/target URI. Previously we erronously only trimmed the '/' from a
'pathless' URI, i.e.:

  https://example.com/      => https://example.com/
  https://example.com/path/ => https://example.com/path/

This is a safe change since the `Enumerate` method that is used by both
`Get` and `Remove` already matches _both_ trailing and non-trailing
slashes by virtue of creating and comparing components of a `System.Uri`
which does normalisation of the path.

Both `Get` and `Remove` would act (eventually) to remove any bad
credentials stored with a trailing slash, even if new credentials are
only written out without the slash.
```